commands/operator-sdk/cmd/build.go

```diff
@@ -41,29 +55,192 @@ For example:
+				if !ok {
+					continue
+				}
+				if image != imageName {
+					warningMessages = fmt.Sprintf("%s\nWARNING: Namespace manifest contains a deployment with image %v, which does not match the name of the image being built: %v", warningMessages, image, imageName)
+				}
+			}
+		}
+	}
+	if warningMessages == "" {
+		return nil
+	}
+	return errors.New(warningMessages)
+}
+
+func renderTestManifest(image string) {
+	namespacedBytes, err := ioutil.ReadFile(namespacedManBuild)
+	if err != nil {
+		log.Fatalf("could not read namespaced manifest: %v", err)
+	}
+	// generate deploy/test-pod.yaml
+	deployDir := filepath.Join(mustGetwd(), "deploy")
+	testPodPath := filepath.Join(deployDir, "test-pod.yaml")
+	testPodfilegen := scaffold.NewTestPodCodegen(&scaffold.TestPodInput{})
+	buf := &bytes.Buffer{}
+	err = testPodfilegen.Render(buf)
+	if err != nil {
+		log.Fatalf("failed to render the template for (%v): %v", testPodPath, err)
+	}
+	err = writeFileAndPrint(testPodPath, buf.Bytes(), defaultFileMode)
+	if err != nil {
+		log.Fatalf("failed to create %v: %v", testPodPath, err)
+	}
+	err = verifyDeploymentImage(namespacedBytes, image)
+	// the error from verifyDeploymentImage is just a warning, not fatal error
+	if err != nil {
+		fmt.Printf("%v\n", err)
+	}
 }
 
 const (
-	build = "./build/build.sh"
+	mainGo = "./cmd/manager/main.go"
 )
 
 func buildFunc(cmd *cobra.Command, args []string) {
 	if len(args) != 1 {
-		log.Fatalf("build command needs 1 argument.")
+		log.Fatalf("build command needs exactly 1 argument")
 	}
 
-	bcmd := exec.Command(build)
-	o, err := bcmd.CombinedOutput()
+	cmdutil.MustInProjectRoot()
+	goBuildEnv := append(os.Environ(), "GOOS=linux", "GOARCH=amd64", "CGO_ENABLED=0")
+	wd, err := os.Getwd()
 	if err != nil {
-		log.Fatalf("failed to build: %v (%v)", string(o), err)
+		log.Fatalf("could not identify current working directory: %v", err)
+	}
+
+	// Don't need to buld go code if Ansible Operator
+	if mainExists() {
+		buildCmd := exec.Command("go", "build", "-o", filepath.Join(wd, "build/_output/bin", filepath.Base(wd)), filepath.Join(cmdutil.GetCurrPkg(), "cmd/manager"))
+		buildCmd.Env = goBuildEnv
+		o, err := buildCmd.CombinedOutput()
+		if err != nil {
+			log.Fatalf("failed to build operator binary: %v (%v)", err, string(o))
+		}
+		fmt.Fprintln(os.Stdout, string(o))
 	}
-	fmt.Fprintln(os.Stdout, string(o))
 
 	image := args[0]
-	dbcmd := exec.Command("docker", "build", ".", "-f", "build/Dockerfile", "-t", image)
-	o, err = dbcmd.CombinedOutput()
+	baseImageName := image
+	if enableTests {
+		baseImageName += "-intermediate"
+	}
+	dbcmd := exec.Command("docker", "build", ".", "-f", "build/Dockerfile", "-t", baseImageName)
+	o, err := dbcmd.CombinedOutput()
 	if err != nil {
-		log.Fatalf("failed to output build image %v: %v (%v)", image, string(o), err)
+		if enableTests {
+			log.Fatalf("failed to build intermediate image for %s image: (%s)", image, string(o))
```

Comment:
Print the err along with the output from `dbcmd.CombinedOutput()`. I know this is carrying over from the master but let's fix this while we can:
```Go
log.Fatalf("failed to build intermediate image for %s image: %v (%s)", image, err, string(o))
```
And below in line 183 as well.